lgc/elfLinker/ElfLinker.cpp

```diff
@@ -409,6 +409,25 @@ bool ElfLinkerImpl::link(raw_pwrite_stream &outStream) {
     }
   }
 
+  // Update the size of the symbols that had code appended to them.
+  for (auto &glueShader : m_glueShaders) {
+    auto glue_name_string_idx = getStringIndex(glueShader->getGlueShaderName());
+    auto glue_shader_sym = findSymbol(glue_name_string_idx);
+    assert(glue_shader_sym != 0);
+
+    auto main_name_string_idx = getStringIndex(glueShader->getMainShaderName());
+    auto main_sym = findSymbol(main_name_string_idx);
+    assert(main_sym != 0);
+
+    if (glueShader->isProlog()) {
+      m_symbols[glue_shader_sym].st_size =
+          (m_symbols[main_sym].st_value + m_symbols[main_sym].st_size) - m_symbols[glue_shader_sym].st_value;
+    } else {
+      m_symbols[main_sym].st_size =
+          (m_symbols[glue_shader_sym].st_value + m_symbols[glue_shader_sym].st_size) - m_symbols[main_sym].st_value;
+    }
+  }
+
```

Comment:
This code relies on the order of `m_glueShaders`, doesn't it? I suspect it's still correct, but could you please take a short note of that in the comment? LGTM apart from that.

Reply:
Good point.  Done.